clang/test/CodeGenSYCL/free_function_int_header.cpp

```diff
@@ -321,39 +369,39 @@ void ff_24(int arg) {
 // CHECK:      const kernel_param_desc_t kernel_signatures[] = {
 // CHECK-NEXT:   {{.*}}__sycl_kernel_ff_2Piii
 // CHECK-NEXT:   { kernel_param_kind_t::kind_pointer, 8, 0 },
-// CHECK-NEXT:   { kernel_param_kind_t::kind_std_layout, 4, 8 },
-// CHECK-NEXT:   { kernel_param_kind_t::kind_std_layout, 4, 12 },
+// CHECK-NEXT:   { kernel_param_kind_t::kind_std_layout, 4, 0 },
```

Comment:
@premanandrao Regarding these changes...

The offsets for arguments of a free function kernel are somewhat meaningless since there is no concept of a closure class as is the case with lambda kernels. Therefore, I think it's less confusing to simply give them an offset of zero and reserve non-zero offsets only in the case of special types that are embedded inside a parameter struct which will help us find the special type inside the struct in the runtime layer. So, all of these zeros are simply offsets of top-level parameters for which a non-zero offset does not make sense in the context of free function kernels.